### PR TITLE
feat: Stubs are created for referenced declarations in other packages

### LINKED ISSUE #66: feat: Stubs must be created for referenced declarations in other packages
### Is your feature request related to a problem?

Currently, if a module imports classes from other packages stubs won't be created for those classes.

Say we are creating stubs for the package `a` and are currently working on a Python module with the following content:

```py
from b import SomeClass

def f(p: SomeClass)
```

Then we also need to create some stub for the class `SomeClass`, or we can't use it. So limited stubs must be created for the declarations of other packages that package `a` depends on.

### Desired solution

Create limited stubs for the imported classes from other packages, if they are imported.

### PR DESCRIPTION
**Pull request by Masara**
Closes #66

### Summary of Changes

Now limited stubs are created for imported classes from other packages.

### CODE REVIEW

**Review by lars-reimann — CHANGES_REQUESTED** [review]

File: tests/safeds_stubgen/stubs_generator/test_generate_stubs.py [hunk 1]
```diff
@@ -68,6 +68,16 @@ def test_file_creation() -> None:
     )
 
 
+def test_file_creation_limited_stubs_outside_package(snapshot_sds_stub: SnapshotAssertion) -> None:
+    # Somehow the stubs get overwritten by other tests, therefore we have to call the function before asserting
```
[the remark this answers is not in the document]
Comment (lars-reimann) [reply tests/safeds_stubgen/stubs_generator/test_generate_stubs.py:72]: This does not necessarily have to be done in this PR, though.

**Comment by Masara** [comment]
@lars-reimann I changed the stubs-generator the way you said and refactored the tests, are the tests like this all right or should I still try to get the file names with ```glob``` like you mentioned [here](https://github.com/Safe-DS/Stub-Generator/pull/69#discussion_r1509919395)?

**Comment by Masara** [comment]
Also, you can merge this PR first.

File: tests/safeds_stubgen/stubs_generator/test_generate_stubs.py [hunk 2]
```diff
@@ -78,50 +74,29 @@ def test_file_creation_limited_stubs_outside_package(snapshot_sds_stub: Snapshot
...
+        "infer_types_module",
+        "import_module",
+        "enum_module",
+        "function_module",
+        "attribute_module",
+        "class_module",
+    ]
+)
+def test_stub_creation(file_name: str, snapshot_sds_stub: SnapshotAssertion) -> None:
+    for stub_data in stubs_data:
```
Comment (lars-reimann) [inline tests/safeds_stubgen/stubs_generator/test_generate_stubs.py:98]: Are you using all entries of `stubs_data`?

**Comment by Masara** [comment]
@lars-reimann I changed the tests, now all stubs files should be checked

**Comment by lars-reimann** [comment]
Yep, looks great this way and should be easier to extend later on.

File: src/safeds_stubgen/stubs_generator/_generate_stubs.py [hunk 3]
```diff
@@ -368,13 +456,15 @@ def _create_property_function_string(self, function: Function, indentations: str
         Functions or methods with the @property decorator are handled the same way as class attributes.
         """
         name = function.name
-        camel_case_name = self._convert_snake_to_camel_case(name)
+        camel_case_name = name
+        if self.convert_identifiers:
+            camel_case_name = _convert_snake_to_camel_case(name)
```
[the remark this answers is not in the document]
Comment (lars-reimann) [reply src/safeds_stubgen/stubs_generator/_generate_stubs.py]: What happened to this one?

[the remark this answers is not in the document]
Comment (Masara) [reply src/safeds_stubgen/stubs_generator/_generate_stubs.py]: I renamed the ```_convert_snake_to_camel_case``` function and added the changed you proposed

[the remark this answers is not in the document]
Comment (Masara) [reply src/safeds_stubgen/stubs_generator/_generate_stubs.py]: Oh nevermind, I misunderstood what you meant, I will add the changes in a moment

[the remark this answers is not in the document]
Comment (lars-reimann) [reply src/safeds_stubgen/stubs_generator/_generate_stubs.py]: It was mostly about moving the if inside the function, so it's always paired with the call.

[the remark this answers is not in the document]
Comment (Masara) [reply src/safeds_stubgen/stubs_generator/_generate_stubs.py]: Yes, I fixed that now and removed the if statements

**Review by lars-reimann — APPROVED** [review]
Nice work!

**Pull request merged**